source/common/tcp_proxy/upstream.h

```diff
@@ -68,16 +85,243 @@ class HttpConnPool : public GenericConnPool, public Http::ConnectionPool::Callba
                    Upstream::HostDescriptionConstSharedPtr host, StreamInfo::StreamInfo& info,
                    absl::optional<Http::Protocol>) override;
 
+  void onUpstreamHostSelected(Upstream::HostDescriptionConstSharedPtr);
+  void onHttpPoolReady(Upstream::HostDescriptionConstSharedPtr& host,
+                       Ssl::ConnectionInfoConstSharedPtr ssl_info);
+
+  struct NullHedgePolicy : public Router::HedgePolicy {
```

Comment:
Nit here and elsewhere please add a brief comment which explains the purpose of this object. Also Router::HedgePolicy is a class, so the subclass should be a class as well (and hence have private member)